build-system/tasks/extension-helpers.js

```diff
@@ -151,6 +151,16 @@ function maybeInitializeExtensions(
   }
 }
 
+/**
+ * Set the extensions to build from example documents
+ * (for internal use by `gulp performance`)
+ *
+ * @param {Array<string>} examples Path to example documents
+ */
+function setExtensionsToBuildFromDocuments(examples) {
+  extensionsToBuild = getExtensionsFromArg(examples.join(','));
```

Comment:
I see two potential problems with this line:
- It excludes the extensions in `DEFAULT_EXTENSION_SET`, which need to be built for all extensions.
- There's the likelihood of duplicate entries for extensions used by more than one example file.

Here's an alternative that should work:

```js
  extensionsToBuild = dedupe(
    DEFAULT_EXTENSION_SET.concat(getExtensionsFromArg(examples.join(',')))
  );
```